packages/shared/src/hooks/useUpdateEmailNotificationSettings.ts

```diff
@@ -0,0 +1,246 @@
+import { useCallback, useMemo, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+
+import { useUpdateEmailNotificationSettingsFragment$key } from '~/generated/useUpdateEmailNotificationSettingsFragment.graphql';
+import { useUpdateEmailNotificationSettingsMutation } from '~/generated/useUpdateEmailNotificationSettingsMutation.graphql';
+
+import { usePromisifiedMutation } from '../relay/usePromisifiedMutation';
+
+type useUpdateEmailNotificationSettingsProps = {
+  queryRef: useUpdateEmailNotificationSettingsFragment$key;
+  hasEarlyAccess: boolean;
+  shouldShowEmailSettings: boolean;
+};
+
+export default function useUpdateEmailNotificationSettings({
+  queryRef,
+  hasEarlyAccess,
+  shouldShowEmailSettings,
+}: useUpdateEmailNotificationSettingsProps) {
+  const query = useFragment(
+    graphql`
+      fragment useUpdateEmailNotificationSettingsFragment on Query {
+        viewer {
+          ... on Viewer {
+            email {
+              emailNotificationSettings {
+                unsubscribedFromNotifications
+                unsubscribedFromDigest
+                unsubscribedFromMarketing
+                unsubscribedFromMembersClub
+              }
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+
+  const currentEmailNotificationSettings = query?.viewer?.email?.emailNotificationSettings;
+
+  const [emailSettings, setEmailSettings] = useState<EmailNotificationSettings>({
+    notifications: !currentEmailNotificationSettings?.unsubscribedFromNotifications,
+    marketing: !currentEmailNotificationSettings?.unsubscribedFromMarketing,
+    digest: !currentEmailNotificationSettings?.unsubscribedFromDigest,
+    membersClub: hasEarlyAccess
+      ? !currentEmailNotificationSettings?.unsubscribedFromMembersClub
+      : false,
+    // currently cannot toggle all notifs from notif setting
+    all: false,
+  });
+
+  const computeToggleChecked = useCallback(
+    (notifType: string | number) => {
+      // if the user dont have an email or not verified, we want to toggle off
+      if (!shouldShowEmailSettings) {
+        return false;
+      }
+      return emailSettings[notifType];
+    },
+    [shouldShowEmailSettings, emailSettings]
+  );
```

Comment:
here we can update L54 to accept this type of input. then we shouldn't have to do the `[key: string]: boolean` trickery below

![image](https://github.com/gallery-so/gallery/assets/12162433/8bff6995-c4e3-42e3-916a-76a9d2a19b34)

Reply:
working now!